corehq/apps/aggregate_ucrs/ingestion.py

```diff
@@ -132,7 +132,8 @@ def populate_aggregate_table_data_for_time_period(aggregate_table_adapter, windo
         sqlalchemy_secondary_table = IndicatorSqlAdapter(secondary_table.data_source).get_table()
         # apply join filters along with period start/end filters (if necessary) for related model
         join_conditions = [
-            primary_table.c['doc_id'] == sqlalchemy_secondary_table.c[secondary_table.data_source_key]
+            (primary_table.c[secondary_table.primary_key_column] ==
```

Comment:
You're assuming here that they have their primary key columns are named the same in the primary table and the secondary and that they refer to the same thing. Is that correct?

Reply:
Ah, I see what this means. `secondary_table.primary_key_column` isn't referring to the primary key of the secondary table, it's referring to the 'key' column of the primary table. I think the name should change since `primary key` is an existing term for a SQL table. Maybe `join_column_on_primary` and `join_column_on_secondary`?

Reply:
thanks, that's way less confusing. went with `join_column_primary`.